### PR TITLE
DMP-2572 CPP daily list fix

### PR DESCRIPTION
**Pull request by davet1985**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-2572

### Change description ###

- ensure the URN is set as part of the Defendant

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/dailylist/mapper/DailyListRequestMapper.java [hunk 1]
```diff
@@ -127,7 +127,20 @@ default List<Defendant> map(HearingStructure.Defendants defendants) {
         return CollectionUtils.emptyIfNull(defendants.getDefendant()).stream().map(this::map).toList();
     }
 
-    Defendant map(DefendantStructure defendant);
+    default Defendant map(DefendantStructure defendant) {
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/dailylist/mapper/DailyListRequestMapper.java:130]: is the same not possible with:-
```
    @Mappings({
        @Mapping(source = "URN", target = "urn")
    })
```

Comment (davet1985) [reply src/main/java/uk/gov/hmcts/darts/dailylist/mapper/DailyListRequestMapper.java]: Yes, that's much better, thanks @cmagowan-hmcts

File: src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java [hunk 2]
```diff
@@ -336,7 +333,7 @@ void testRoutesInvalidDailyListPayloadNoDartsAPIProblemBody(
 
     @ParameterizedTest
     @ArgumentsSource(DartsClientProvider.class)
-    void removesLineBreaksFromDailyListXml(
+    void testCppDailyListXmlWithLineBreaks(
```
Comment (tonymort) [inline src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java:336]: Isn't it better to specifiy what we are testing?

File: src/integrationTest/resources/payloads/events/valid-dailyList-with-line-breaks.xml [hunk 3]
```diff
@@ -26,7 +26,7 @@
     &lt;cs:CourtList&gt;
     &lt;cs:CourtHouse&gt;
     &lt;cs:CourtHouseType&gt;Crown Court&lt;/cs:CourtHouseType&gt;
-    &lt;cs:CourtHouseCode&gt;467&lt;/cs:CourtHouseCode&gt;
+    &lt;cs:CourtHouseCode>467&lt;/cs:CourtHouseCode&gt;
```
Comment (cmagowan-hmcts) [inline src/integrationTest/resources/payloads/events/valid-dailyList-with-line-breaks.xml:29]: looks wrong?

**Review by tonymort — APPROVED** [review]
Just the 1 minor comment otherwise all good

File: src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java (hunk 2 above)
Comment (davet1985) [reply src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java:336]: I've made it slightly more general, what name do you think @tonymort ?

File: src/integrationTest/resources/payloads/events/valid-dailyList-with-line-breaks.xml (hunk 3 above)
Comment (davet1985) [reply src/integrationTest/resources/payloads/events/valid-dailyList-with-line-breaks.xml]: Removed these changes

**Review by cmagowan-hmcts — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java (hunk 2 above)
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java:336]: I think if we've had a bug and a test has been written to cover the scenario of the bug it should remain a specific test. So if theres another bug personally I'd create another test named specifically for that bug.

Comment (davet1985) [reply src/integrationTest/java/uk/gov/hmcts/darts/ws/EventWebServiceTest.java]: Added a specific test as discussed 👍

**Pull request merged**